utilities/src/main/java/io/syndesis/qe/resource/impl/Syndesis.java

```diff
@@ -163,6 +150,34 @@ public void createPullSecret() {
         }
     }
 
+    private void jaegerWorkarounds() {
+        // the syndesis-jaeger doesn't contain "syndesis.io/component" label which is using for finding all components. It is added manually here
+        new Thread(() -> {
+            int tries = 5;
+            while (tries > 0) {
+                try {
+                    OpenShiftWaitUtils.waitFor(() -> OpenShiftWaitUtils.isPodReady(
+                        OpenShiftUtils.getAnyPod("app.kubernetes.io/instance", "syndesis-jaeger")),
+                        30 * 1000);
+
+                    break;
+                } catch (Exception e) {
+                    ResourceFactory.get(Jaeger.class).ensureImagePull();
+                    log.info("Ensuring jaeger instance is pulled correctly, linking secrets and resetting jaeger pods");
+                }
+                tries--;
+            }
+            Optional<Pod> jaegerPod = OpenShiftUtils.getPodByPartialName("syndesis-jaeger");
```

Comment:
The pod may not be ready in case the while cycle finished with tries==0 . The `ensureImagePull()` remove the Jaeger pod and the program execution continues so the `jaegerPod.isPresent()` returns false. After a while cycle, it needs to wait until the Jaeger is back to ready.